cmd/timescale-prometheus/main.go

```diff
@@ -31,25 +32,28 @@ import (
 )
 
 type config struct {
-	listenAddr        string
-	telemetryPath     string
-	pgmodelCfg        pgclient.Config
-	logLevel          string
-	haGroupLockID     int
-	restElection      bool
-	prometheusTimeout time.Duration
-	electionInterval  time.Duration
-	migrate           bool
-	stopAfterMigrate  bool
-	corsOrigin        *regexp.Regexp
+	listenAddr         string
+	telemetryPath      string
+	pgmodelCfg         pgclient.Config
+	logLevel           string
+	haGroupLockID      int64
+	restElection       bool
+	prometheusTimeout  time.Duration
+	electionInterval   time.Duration
+	migrate            bool
+	stopAfterMigrate   bool
+	lockWhileMigrating bool
+	corsOrigin         *regexp.Regexp
 }
 
 const (
 	promLivenessCheck = time.Second
+	schemaLockId      = 0x4D829C732AAFCEDE // chosen randomly
```

Comment:
nit: should we move this into util/lock.go?

Reply:
I don't think so. `lock.go` describes the abstract locking mechanism, but defining which lock we use it up to the application. (if anything I'm inclined to move `PgLeaderLock` to `election.go`, or else rename it to describe what it is instead of how it's used, but I can't think of a goon name, and this PR is large enough already)